instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/AwsLambdaTracer.java

```diff
@@ -69,13 +89,37 @@ private SpanBuilder createSpan(Context context, Object input, Map<String, String
   }
 
   private void setAttributes(SpanBuilder span, Context context, Object input) {
-    span.setAttribute(FAAS_EXECUTION, context.getAwsRequestId());
+    setCommonAttributes(span, context);
     if (input instanceof APIGatewayProxyRequestEvent) {
       span.setAttribute(FAAS_TRIGGER, FaasTriggerValues.HTTP.getValue());
       httpSpanAttributes.onRequest(span, (APIGatewayProxyRequestEvent) input);
     }
   }
 
+  private void setCommonAttributes(SpanBuilder span, Context context) {
+    span.setAttribute(FAAS_EXECUTION, context.getAwsRequestId());
+    String arn = getFunctionArn(context);
+    if (arn != null) {
+      span.setAttribute(FAAS_ID, arn);
+      String[] arnParts = arn.split(":");
+      if (arnParts.length >= 5) {
+        span.setAttribute(CLOUD_ACCOUNT_ID, arnParts[4]);
```

Comment:
Since account ID is constant, how about storing it into a lazy-inited field? We probably don't need to make it volatile since it'll converge eventually.

Reply:
I can store the value (whole ARN actually) in a lazy field, but is there any point in doing this? I mean, right now this code will only execute once per whole lambda so it won't be reused anywhere.

Reply:
Isn't it called once per span, meaning once per lambda invocation? And I think by caching we wouldn't need to recompute on every invocation.

Reply:
> You can add initialization code outside of your handler method to reuse resources across multiple invocations. When the runtime loads your handler, it runs static code and the class constructor. Resources that are created during initialization stay in memory between invocations, and can be reused by the handler thousands of times.

😮 
For some reason I was convinced that the whole JVM gets initialized for each lambda invocation. OK, now everything makes sense -- I'll cache it, that's a very good suggestion. Thanks!

Reply:
That'd be a pretty crazy slow runtime ;)